spec/mem/tree_borrows/tree.md

```diff
@@ -31,7 +31,7 @@ Then we can define the node. Structurally, we use the usual functional represent
 struct Node {
     children: List<Node>,
     /// State for each location
-    location_states: List<LocationState>,
+    permissions: List<Permission>,
     /// Indicates whether the node is protected by a function call,
     /// i.e., whether the original reference passed as an argument of a function call.
     /// This will be some kind of protection (weak or strong) if and only if this node is in
```

Comment:
As your PR description explains, the `protected` field is now partially redundant. Can you state the invariant that must always hold, and maybe even assert it in some suitable place in the code?